### PR TITLE
Refactored examples to not use deprecated methods

### PR DESCRIPTION
**Pull request by zarthross**
Refactored examples to not use deprecated methods and also refactored them so the entry points look basically the same.

### CODE REVIEW

File: examples/src/main/scala/com/http4s/rho/swagger/demo/Main.scala [hunk 1]
```diff
@@ -1,15 +1,28 @@
...
 import org.http4s.server.blaze.BlazeBuilder
+import org.http4s.server.{Server, ServerApp}
+import org.log4s.getLogger
+
+import scalaz.concurrent.Task
+
+object Main extends ServerApp {
+  private val logger = getLogger
+
+  val port = Option(System.getenv("HTTP_PORT"))
```
Comment (bryce-anderson) [inline examples/src/main/scala/com/http4s/rho/swagger/demo/Main.scala]: Nice touch :+1:

**Comment by bryce-anderson** [comment]
Thanks @zarthross!

File: examples/src/main/scala/com/http4s/rho/swagger/demo/Main.scala (hunk 1 above)
Comment (zarthross) [inline examples/src/main/scala/com/http4s/rho/swagger/demo/Main.scala]: Can't take the credit for that one, it was lifted directly from the existing Hal example.  Glad you like it though.

**Comment by zarthross** [comment]
No problem!

**Pull request merged**